lib/Backends/Interpreter/InterpreterNodes.cpp

```diff
@@ -396,54 +402,58 @@ void InterpreterFunction::fwdAvgPoolInst(const AvgPoolInst *I) {
 
-  if (I->getSrc()->getType()->isQuantizedType()) {
-    auto inW = getWeightHandle<int8_t>(I->getSrc());
-    auto outW = getWeightHandle<int8_t>(I->getDest());
-    TensorQuantizationParams inQP{I->getSrc()->getType()->getScale(),
-                                  I->getSrc()->getType()->getOffset()};
-    TensorQuantizationParams outQP{I->getDest()->getType()->getScale(),
-                                   I->getDest()->getType()->getOffset()};
-
-    // For each input in the batch:
-    for (size_t n = 0; n < odim.n; n++) {
-      // For each layer in the output tensor:
-      for (size_t z = 0; z < idim.c; z++) {
-        // For each convolution 'jump' in the input tensor:
-        ssize_t x = -ssize_t(pdim.top);
-        for (size_t ax = 0; ax < odim.h; x += sdim.height, ax++) {
-          ssize_t y = -ssize_t(pdim.left);
-          for (size_t ay = 0; ay < odim.w; y += sdim.width, ay++) {
-            int32_t sum = 0;
+  auto inW = getWeightHandle<ElemTy>(I->getSrc());
+  auto outW = getWeightHandle<ElemTy>(I->getDest());
 
-            for (size_t fx = 0; fx < kdim.height; fx++) {
-              for (size_t fy = 0; fy < kdim.width; fy++) {
-                ssize_t ox = x + fx;
-                ssize_t oy = y + fy;
+  // For each input in the batch:
+  for (size_t n = 0; n < odim.n; n++) {
+    // For each layer in the output tensor:
+    for (size_t z = 0; z < idim.c; z++) {
+      // For each convolution 'jump' in the input tensor:
+      ssize_t x = -ssize_t(pdim.top);
+      for (size_t ax = 0; ax < odim.h; x += sdim.height, ax++) {
+        ssize_t y = -ssize_t(pdim.left);
+        for (size_t ay = 0; ay < odim.w; y += sdim.width, ay++) {
+          float sum = 0;
 
-                // Ignore index access below zero (this is due to padding).
-                if (ox < 0 || oy < 0 || ox >= ssize_t(idim.h) ||
-                    oy >= ssize_t(idim.w)) {
-                  continue;
-                }
+          for (size_t fx = 0; fx < kdim.height; fx++) {
+            for (size_t fy = 0; fy < kdim.width; fy++) {
+              ssize_t ox = x + fx;
+              ssize_t oy = y + fy;
 
-                sum += inW.at({n, (size_t)ox, (size_t)oy, z}) - inQP.offset;
+              // Ignore index access below zero (this is due to padding).
+              if (ox < 0 || oy < 0 || ox >= ssize_t(idim.h) ||
+                  oy >= ssize_t(idim.w)) {
+                continue;
               }
+
+              sum += float(inW.at({n, (size_t)ox, (size_t)oy, z}));
             }
-            // Instead of dividing by filterArea, just change scale.
-            outW.at({n, ax, ay, z}) =
-                quantization::clip<int32_t, int8_t>(std::round(
-                    float(sum) * (inQP.scale / outQP.scale / filterArea) +
-                    outQP.offset));
-          } // W
-        }   // H
-      }     // C
-    }       // N
+          }
+          outW.at({n, ax, ay, z}) = ElemTy(sum / filterArea);
```

Comment:
Thanks @jfix71 to point this out.

It could make a difference but so far we decide to do all the computation in fp32 and cast the result only at the end.
There isn’t a good or bad answer here and the results will be different from target to target because of implementation choices and fast math optimization.